include/WCSimPrimaryGeneratorAction.hh

```diff
@@ -23,23 +26,37 @@ public:
   void GeneratePrimaries(G4Event* anEvent);
 
   // Normal gun setting calls these functions to fill jhfNtuple and Root tree
-  void SetVtx(G4ThreeVector i)     { vtx = i; };
-  void SetBeamEnergy(G4double i)   { beamenergy = i; };
-  void SetBeamDir(G4ThreeVector i) { beamdir = i; };
-  void SetBeamPDG(G4int i)         { beampdg = i; };
+  void SetVtx(G4ThreeVector i)     { vtxs[0] = i; nvtxs = 1; };
+  void SetBeamEnergy(G4double i)   { beamenergies[0] = i;};
+  void SetBeamDir(G4ThreeVector i) { beamdirs[0] = i;};
+  void SetBeamPDG(G4int i)         { beampdgs[0] = i;};
+  void SetNvtxs(G4int i)     { nvtxs = i; };
+  void SetVtxs(G4int i, G4ThreeVector v)     { vtxs[i] = v; };
```

Comment:
To make it more user friendly & clearer to use, I think removing two methods that do the same thing is a good thing to do. So remove the Set/Get methods that are plural, and add a G4int n = 0 argument to the non-plural methods.